src/ucp/core/ucp_request.h

```diff
@@ -60,6 +61,7 @@ typedef struct ucp_send_state {
  * Request in progress.
  */
 struct ucp_request {
+    ucs_status_t                  status;  /* Operation status */
```

Comment:
I've made several experiments/micro-benchmarking with it, and found no performance implications from moving this into the end of struct. So finally I added it here close to the flag.